packages/react/src/useSignUpSecurityKeyEmail.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  SignUpSecurityKeyOptions,
+  signUpSecurityKeyPromise,
+  SignUpSecurityKeyState
+} from '@nhost/core'
+import { useSelector } from '@xstate/react'
+
+import { useAuthInterpreter } from './useAuthInterpreter'
+
+type SignUpSecurityKeyHandlerResult = Omit<SignUpSecurityKeyState, 'isLoading'>
+
+interface SignUpSecurityKeyHandler {
+  (email: string, options?: SignUpSecurityKeyOptions): Promise<SignUpSecurityKeyHandlerResult>
+}
+
+export interface SignUpSecurityKeyHookResult extends SignUpSecurityKeyState {
+  /** Used for a new user to sign up with a security key. Returns a promise with the current context */
+  signUpSecurityKeyEmail: SignUpSecurityKeyHandler
+}
+
+interface SignUpSecurityKeyHook {
+  (options?: SignUpSecurityKeyOptions): SignUpSecurityKeyHookResult
+}
+
+/**
+ * Use the hook `useSignUpSecurityKey` to sign up a user with security key and an email using the WebAuthn API.
+ *
+ * @example
+ * ```tsx
+ * const { signUpSecurityKey, needsEmailVerification, isLoading, isSuccess, isError, error } = useSignUpSecurityKey()
```

Comment:
comment hook name is `useSignUpSecurityKey` but the actual name is `useSignUpSecurityKeyEmail`.

I think it should be `useSignUpEmailSecurityKey`.

Reply:
According to the comment docs here it's possible to get `isLoading, isSuccess`, etc from the hook.

It is possible to get these state variables both from the hook and also later from `signUpSecurityKey`?

Reply:
> According to the comment docs here it's possible to get `isLoading, isSuccess`, etc from the hook.
> 
> It is possible to get these state variables both from the hook and also later from `signUpSecurityKey`?

yes, except `isLoading` as `signUpSecurityKey` is a promise that is resolved when loading is done